Monika After Story/game/definitions.rpy

```diff
@@ -707,22 +712,52 @@ python early:
             # setup lock entry
             Event.INIT_LOCKDB.setdefault(eventlabel, mas_init_lockdb_template)
 
+            # Cache conditional
+            if self.conditional is not None and self.conditional not in Event._conditional_cache:
+                Event._conditional_cache[self.conditional] = renpy.python.py_compile(self.conditional, "eval")
 
-        # equality override
         def __eq__(self, other):
+            """
+            Equality override
+
+            IN:
+                other - the object to compare this event with
+
+            OUT:
+                boolean
+            """
             if isinstance(self, other.__class__):
                 return self.eventlabel == other.eventlabel
             return False
 
-        # equality override
         def __ne__(self, other):
+            """
+            Non-equality override
+
+            IN:
+                other - the object to compare this event with
+
+            OUT:
+                boolean
+            """
             return not self.__eq__(other)
 
-        # set attr overrride
         def __setattr__(self, name, value):
-            #
-            # Override of setattr so we can do cool things
-            #
+            """
+            Override of setattr so we can do cool things
+
+            IN:
+                name - the name of the prop to change (str)
+                value - the new value
+            """
+            # Special handling for the conditional property
+            if (
+                name == "conditional"
+                and value is not None
+                and value not in Event._conditional_cache
+            ):
+                Event._conditional_cache[value] = renpy.python.py_compile(value, "eval")
```

Comment:
`conditional` is not part of `N_EVENT_NAMES`, so we could move this to the `else` block of the that if statement and save a branch check. Although specifically you should move it to the `attr_loc` check after the `start_date` `end_date` checks.